src/coreclr/utilcode/executableallocator.cpp

```diff
@@ -905,7 +905,7 @@ void* ExecutableAllocator::MapRW(void* pRX, size_t size, CacheableMapping cacheM
 
             if (pRW == NULL)
             {
-                g_fatalErrorHandler(COR_E_EXECUTIONENGINE, W("Failed to create RW mapping for RX memory"));
+                g_fatalErrorHandler(COR_E_EXECUTIONENGINE, W("Failed to create RW mapping for RX memory. This can be caused by not enough memory or by exceeding the configured maximum number of memory mappings."));
```

Comment:
```suggestion
                g_fatalErrorHandler(COR_E_EXECUTIONENGINE, W("Failed to create RW mapping for RX memory. This can cause by insufficient memory or hitting the limit of memory mapping on Unix (vm.map_max_count)."));
```
```suggestion
                g_fatalErrorHandler(COR_E_EXECUTIONENGINE, W("Failed to create RW mapping for RX memory. This can be caused by not enough memory or by exceeding the configured maximum number of memory mappings."));
```